src/cli/parse-flags.ts

```diff
@@ -35,203 +54,197 @@ export const parseFlags = (args: string[], sys?: CompilerSystem): ConfigFlags =>
   }
 
   flags.unknownArgs = flags.args.filter((arg: string) => {
-    return !flags.knownArgs.includes(arg);
+    return !flags.knownArgs!.includes(arg);
   });
 
   return flags;
 };
 
 /**
- * Parse command line arguments that are whitelisted via the BOOLEAN_ARG_OPTS,
- * STRING_ARG_OPTS, and NUMBER_ARG_OPTS arrays in this file. Handles leading
- * dashes on arguments, aliases that are defined for a small number of argument
- * types, and parsing values for non-boolean arguments (e.g. port number).
+ * Parse command line arguments that are enumerated in the BOOLEAN_ARG_OPTS,
+ * STRING_ARG_OPTS, NUMBER_ARG_OPTS, and LOG_LEVEL_OPTS arrays in the
+ * `config-flags` module. Handles leading dashes on arguments, aliases that
+ * are defined for a small number of argument types, and parsing values for
+ * non-boolean arguments (e.g. port number).
  *
- * @param flags     a ConfigFlags object
- * @param args      an array of command-line arguments to parse
- * @param knownArgs an array to which all recognized, legal arguments are added
+ * @param flags a ConfigFlags object to which parsed arguments will be added
+ * @param args  an array of command-line arguments to parse
  */
-const parseArgs = (flags: any, args: string[], knownArgs: string[]) => {
-  BOOLEAN_ARG_OPTS.forEach((booleanName) => {
-    const alias = ARG_OPTS_ALIASES[booleanName];
-    const flagKey = configCase(booleanName);
+const parseArgs = (flags: ConfigFlags, args: string[]) => {
+  BOOLEAN_CLI_ARGS.forEach((argName) => parseBooleanArg(flags, args, argName));
+  STRING_CLI_ARGS.forEach((argName) => parseStringArg(flags, args, argName));
+  NUMBER_CLI_ARGS.forEach((argName) => parseNumberArg(flags, args, argName));
+  LOG_LEVEL_CLI_ARGS.forEach((argName) => parseLogLevelArg(flags, args, argName));
```

Comment:
here I created a separate arg parsing function for each support arg type (boolean, string, etc). I was able to share some logic between some of them (see the `getValue`and `getEqualsValue` function below) and ensure that everything is typed correctly (no `any`)

Reply:
the actual logic for how arguments are parsed should be the same as before the refactor, the test suite passes 100% without changing any tests (I added a few and added some more checks to one or two)